### ISSUE TITLE
unawaited_futures: False positives -> '.then'

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/linter#534]
This should not trigger a lint warning:

```dart
new File('someFile.txt').readAsString().then(print);
```

But it does. On top of it, I can't ignore the lint warning at all:

```dart
// ignore: unawaited_futures
new File('someFile.txt').readAsString().then(print);
```

If this can't be fixed I'm request the lint be disabled for the next SDK release.

**Comment by matanlurey** [comment]
_EDIT_: Looks like restarting analysis does "fix" it, temporarily. New warnings are not ignorable until you restart again, though, and it sort of still presents itself as you code/add more.

**Comment by matanlurey** [comment]
Ping.

This is really annoying. I'd prefer this fixed before new lints are added if that's reasonable.

**Comment by alexeieleusis** [comment]
The new lints you are seeing are part of a project trying to add the
readability/effective dart related lints. Once they are added (the machine
verifiable ones of course) some other improvements will be triaged, but so
far we are planning on adding support for those rules in the analysis
server to have the corresponding quick fixes available in intellij and
other editors.

Since this project mostly lives of community contributions, it is a good
idea to reach directly to the author, I believe it to be @ochafik.

Alexei Diaz | Software Engineer | alexeidiaz@google.com | (650) 214-6432

**Comment by matanlurey** [comment]
Where is the quick fix support? That's part of the linter?

**Comment by bwilkerson** [comment]
Quick fixes lives in the analysis_server package.

**Comment by natebosch** [comment]
@bwilkerson - is there a reasonable chance of getting a quickfix that wraps the future with a call to `unawaited` from `package:pedantic`?

**Comment by bwilkerson** [comment]
I have no problem having such a fix added; it's merely a question of priorities and resources.

**Comment by matanlurey** [comment]
I haven't seen an issue with this.
